verification/src/http_server/handlers/verification/mod.rs

```diff
@@ -1,10 +1,71 @@
-use serde::Serialize;
+#![allow(dead_code)]
+
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
 
 pub mod routes;
 pub mod solidity;
 pub mod sourcify;
 
-#[derive(Debug, Serialize, PartialEq)]
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub struct ContractLibrary {
+    pub lib_name: String,
+    pub lib_address: String,
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
 pub struct VerificationResponse {
-    pub verified: bool,
+    pub message: String,
+    pub result: Option<VerificationResult>,
+    pub status: VerificationStatus,
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub struct VerificationResult {
+    pub contract_name: String,
+    pub compiler_version: String,
+    pub evm_version: String,
+    pub constructor_arguments: Option<String>,
+    pub contract_libraries: Option<Vec<ContractLibrary>>,
+    pub abi: String,
+    pub sources: HashMap<String, String>,
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub enum VerificationStatus {
+    #[serde(rename = "0")]
+    Ok,
+    #[serde(rename = "1")]
+    BytecodeDismatch,
+    #[serde(rename = "99")]
+    UnknowError,
+}
```

Comment:
perhaps this approach is a little more clear, however it is much larger and it seems that it is not really needed, moreover we have to repeat error code two times, it may cause strange bugs

Reply:
It seems to be beneficial from two points of view:
1. If new errors are added, we are enforced by the compiler to process new errors during serialization process, while by `serde(rename)` it is not required (and we are allowed not to specify `serde(rename)`)
2. If we decide to change the type of the status (e.g. to be an uint), it will be impossible to do with current approach

I agree, that error codes duplication may cause some bugs, but in general, seems that `Deserialization` is required only for testing purposes and is not going to be used in production purposes. Thus, the greatest problem it may cause is not to let tests pass. But if that happens, we may easily understand that we forgot to update `TryFrom<String> for VerificationStatus`

Reply:
@leviska What do you think?

Reply:
Sorry, forgot two directives:
```
#[serde(into = "String")]
#[serde(try_from = "String")]
```
Otherwise, it works
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2fcce260c1e89a12e456828e749878fa